src/django_components/component.py

```diff
@@ -843,3 +830,35 @@ def _template_render(self: Template, context: Context, *args: Any, **kwargs: Any
 
     # See https://stackoverflow.com/a/42154067/9788634
     template.render = types.MethodType(_template_render, template)
+
+
+@contextmanager
+def _maybe_bind_template(context: Context, template: Template) -> Generator[None, Any, None]:
```

Comment:
The body of `_maybe_bind_template` is actually what fixes https://github.com/EmilStenstrom/django-components/issues/634.